storage/blob_test.go

```diff
@@ -1022,3 +1022,8 @@ func randBytes(n int) []byte {
 	}
 	return data
 }
+
+func randName(n int) string {
+	name := randString(n) + "/" + randString(n) + "-._~:?#[]@!$&'()*,;= " + randString(n)
```

Comment:
we probably don't want to have `"-._~:?#[]@!$&'()*,;= "` in all tests. That'll make debugging very hard. Maybe have a method like `randNameWithSpecialChars` and invoke this from a new test method called `TestPutBlobWithSpecialName`?